conman/routes/managers.py

```diff
@@ -30,3 +32,31 @@ def best_match_for_path(self, path):
         except IndexError:
             msg = 'No matching Route for URL. (Have you made a root Route?)'
             raise self.model.DoesNotExist(msg)
+
+    def move_branch(self, old_url, new_url):
+        """
+        Move this Route and all descendants to a new URL.
+
+        Movement will be atomic. This means that failure to move any one Route
+        will cause all movement to fail. A conflicting URL will cause an
+        IntegrityError.
+        """
+        to_move = self.filter(url__startswith=old_url)
+        to_move.update(url=Concat(
+            Value(new_url),
+            Substr('url', len(old_url) + 1),  # 1 indexed
+        ))
+
+    def swap_urls(self, path_a, path_b):
+        """
+        Swap the urls of two Routes without moving their descendants.
```

Comment:
> Should the paths be considered implementation details of Routes in this context or are they really the core concept here?

I'd say they're key to the concept, really.

> it's much more important how these are going to be used.

Yeah, that's the angle I was coming from. I'm primarily concerned with making the API logical.

I'm expecting to make use of these methods when I come to implementing custom admin actions/views, but would also like to make them accessible to users of the library.